go.mod

```diff
@@ -1,73 +1,87 @@
 	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
 	github.com/modern-go/reflect2 v1.0.2 // indirect
+	github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822 // indirect
 	github.com/oklog/run v1.1.0 // indirect
+	github.com/pkg/browser v0.0.0-20210911075715-681adbf594b8 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
-	github.com/spf13/cobra v1.4.0 // indirect
-	github.com/stretchr/objx v0.2.0 // indirect
-	golang.org/x/crypto v0.0.0-20220411220226-7b82a4e95df4 // indirect
-	golang.org/x/net v0.0.0-20220225172249-27dd8689420f // indirect
-	golang.org/x/sys v0.0.0-20220412211240-33da011f77ad // indirect
+	github.com/spf13/afero v1.9.2 // indirect
+	github.com/spf13/cobra v1.5.0 // indirect
+	github.com/stretchr/objx v0.4.0 // indirect
+	golang.org/x/crypto v0.0.0-20220926161630-eccd6366d1be // indirect
+	golang.org/x/net v0.0.0-20221002022538-bcab6841153b // indirect
+	golang.org/x/oauth2 v0.0.0-20220909003341-f21342109be1 // indirect
+	golang.org/x/sys v0.0.0-20220928140112-f11e5e49a4ec // indirect
+	golang.org/x/term v0.0.0-20220919170432-7a66f970e087 // indirect
 	golang.org/x/text v0.3.7 // indirect
-	google.golang.org/genproto v0.0.0-20220107163113-42d7afdf6368 // indirect
-	google.golang.org/grpc v1.42.0 // indirect
-	google.golang.org/protobuf v1.27.1 // indirect
+	golang.org/x/time v0.0.0-20220922220347-f3bd1da661af // indirect
+	google.golang.org/appengine v1.6.7 // indirect
+	google.golang.org/genproto v0.0.0-20220930163606-c98284e70a91 // indirect
+	google.golang.org/grpc v1.49.0 // indirect
+	google.golang.org/protobuf v1.28.1 // indirect
 	gopkg.in/inf.v0 v0.9.1 // indirect
 	gopkg.in/yaml.v2 v2.4.0 // indirect
-	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
-	k8s.io/klog/v2 v2.60.1 // indirect
-	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9 // indirect
-	sigs.k8s.io/json v0.0.0-20211208200746-9f7c6b3444d2 // indirect
-	sigs.k8s.io/structured-merge-diff/v4 v4.2.1 // indirect
+	gopkg.in/yaml.v3 v3.0.1 // indirect
+	k8s.io/client-go v0.25.2 // indirect
+	k8s.io/klog/v2 v2.80.1 // indirect
+	k8s.io/kube-openapi v0.0.0-20220928191237-829ce0c27909 // indirect
+	k8s.io/utils v0.0.0-20220922133306-665eaaec4324 // indirect
+	sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
+	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
-replace (
-	github.com/gogo/protobuf => github.com/gogo/protobuf v1.3.2
-	golang.org/x/crypto => golang.org/x/crypto v0.0.0-20201216223049-8b5274cf687f
-)
+// fixes:
+// cloud.google.com/go/compute/metadata: ambiguous import: found package cloud.google.com/go/compute/metadata in multiple modules:
+// cloud.google.com/go v0.97.0 (/go/pkg/mod/cloud.google.com/go@v0.97.0/compute/metadata)
+// cloud.google.com/go/compute v1.7.0 (/go/pkg/mod/cloud.google.com/go/compute@v1.7.0/metadata)
+exclude cloud.google.com/go/compute v1.7.0
```

Comment:
I still cannot develop the code with the IDE Goland, it reports the error
```
GOROOT=/usr/local/go #gosetup
GOPATH=/Users/yinw/go #gosetup
/usr/local/go/bin/go list -modfile=/Users/yinw/go/src/github.com/vmware-tanzu/velero-plugin-for-microsoft-azure/go.mod -m -json -mod=mod all #gosetup
go: k8s.io/kubectl@v0.0.0: invalid version: unknown revision v0.0.0
```
This should be caused by the [kubectl import in azuredisk-csi-driver](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/v1.23.0/go.mod#L108),  adding the follow replace instruction can workaround:
```
replace (
       k8s.io/kubectl => k8s.io/kubectl v0.25.2
)
```

Reply:
And for the `exclude cloud.google.com/go/compute v1.7.0` part, I'm suggesting using replace instead:
```
replace (
       cloud.google.com/go => cloud.google.com/go v0.104.0
)
```
Version 0.104.0 enables the go module for the `compute` package.

Reply:
Added like you suggested.